### ISSUE TITLE
A Spring Boot + Spring Security app with no SecurityConfiguration has infinite redirect

### ISSUE THREAD

**Issue by mraible** [stormpath/stormpath-sdk-java#1236]
If I don't specify a `SecurityConfiguration` class and have stormpath and spring-security in my classpath, there's an infinite redirect.

![screen shot 2017-01-16 at 5 13 06 pm](https://cloud.githubusercontent.com/assets/17892/22026948/7bee8390-dc8f-11e6-829e-0ce2eb140f00.png)

Seems like we should be able to detect this and 1) auto-register an empty `SecurityConfiguration` or 2) print a warning to the console.

Steps to reproduce: 

1. Create a new project.

```
http https://start.stormpath.io/starter.zip dependencies==data-jpa,data-rest,h2,web,devtools,security,stormpath -d
```

2. Expand `demo.zip` and run it using `mvn spring-boot:run`. 
3. Navigate to http://localhost:8080 and witness infinite redirect.

**Comment by mrioan** [comment]
I think https://github.com/stormpath/stormpath-sdk-java/issues/1254 is a duplicate of this one. Please check if the fix for this also fixes 1254

**Comment by mraible** [comment]
@mrioan What do you mean by "the fix for this"? I'm not using IDSite, so I don't think it's related. It's  new project with no code or config for Stormpath, just a dependency.

**Comment by dogeared** [comment]
check waffle. #1250 resolves this issue.

**Comment by mrioan** [comment]
@mraible, @dogeared, I think that #1254 is another scenario that share the same root cause for this issue. The  IDsite callback is closed and therefore a re-login is happening one after the other. So that issue seems to be similar to this one. In both of them an a loops is happening. 

A non-SecurityConfiguration means that there is no explicit open for `/` so it seems a similar root cause to #1254. I am not saying they are indeed duplicates but I think they are sharing the same root cause.

When this issue is implemented the developer should also check if #1254 is also fixed. I have the suspicion it till be fixed. Makes sense?

**Comment by dogeared** [comment]
@mrioan - I will see if #1250 resolves #1254 as well
